error_translator.go

```diff
@@ -7,6 +7,7 @@ import (
 )
 
 var errCodes = map[int]error{
+	1555: gorm.ErrDuplicatedKey,
```

Comment:
This is not DuplicatedKey violation error, checkout this https://www.sqlite.org/rescode.html#constraint_primarykey

Reply:
How about other databases, do they have any error code like `CONSTRAINT_PRIMARYKEY`, if not we can use the same one.